In [ ]:
print("hello ... ")

hello ... 


In [3]:
a = 10
if a > 1:
    print(1)

1


In [6]:
import numpy

In [ ]:
cost = [1200, 1500, 800, 600, 1000, 400, 900]

In [1]:
import numpy as np

In [2]:
arr = np.array([1, 2, 3, 4, 5])
arr 

array([1, 2, 3, 4, 5])

In [3]:
print(arr)

[1 2 3 4 5]


In [4]:
arr2 = np.arange(1, 10, 2)
print(arr2)

[1 3 5 7 9]


In [ ]:
np.int

In [12]:
import numpy as np

arr3 = np.array([1, 2, 127, 5000], dtype=np.int8)
print(arr3)

OverflowError: Python integer 5000 out of bounds for int8

In [15]:
arr4 = np.random.randint(1, 100, 10)
arr4

array([ 3, 36, 13, 28,  9, 48, 62, 43, 60, 79], dtype=int32)

In [16]:
arr7 = np.full(10, 7)
arr7


array([7, 7, 7, 7, 7, 7, 7, 7, 7, 7])

In [18]:
arr8 = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
arr8

array([[1, 2, 3],
       [4, 5, 6],
       [7, 8, 9]])

In [19]:
np.sort(arr4)


array([ 3,  9, 13, 28, 36, 43, 48, 60, 62, 79], dtype=int32)

In [20]:
arr4[arr4 < 20]

array([ 3, 13,  9], dtype=int32)

In [21]:
arr8

array([[1, 2, 3],
       [4, 5, 6],
       [7, 8, 9]])

In [22]:
arr8 = arr8.flatten()
arr8

array([1, 2, 3, 4, 5, 6, 7, 8, 9])

In [23]:
arr8 = arr8.reshape(3, 3)
arr8


array([[1, 2, 3],
       [4, 5, 6],
       [7, 8, 9]])

In [29]:
arr12 = np.random.randint(0, 101, size = 10)
print(arr12)
min_number = np.min(arr12)
print(min_number)
max_number = np.max(arr12)
print(max_number)
averege_number = np.mean(arr12)
print(averege_number)
np.sort(arr12)

[36 79 11 35 18  0 39 87 49 23]
0
87
37.7


array([ 0, 11, 18, 23, 35, 36, 39, 49, 79, 87], dtype=int32)

In [30]:
import pandas as pd
import numpy as np

In [34]:
transaction_df = pd.read_csv('https://raw.githubusercontent.com/dmitriy-shtefan/python-ai/main/data/transactions.csv')

In [35]:
transaction_df

,transaction_id,transaction_date,customer_id,city,category,payment_method,amount,discount_percent
0,1,2026-05-01,C001,Kyiv,Books,Card,420.5,5.0
1,2,2026-05-01,C002,Lviv,Electronics,Card,3299.0,10.0
2,3,2026-05-02,C003,Odesa,Groceries,Cash,256.7,0.0
3,4,2026-05-02,C004,Kharkiv,Clothing,Card,1180.0,15.0
4,5,2026-05-03,C005,Kyiv,Groceries,Card,315.2,0.0
5,6,2026-05-03,C006,Dnipro,Books,Cash,189.9,NaN
6,7,2026-05-04,C007,Lviv,Clothing,Card,950.0,20.0
7,8,2026-05-04,C008,Kyiv,Electronics,Transfer,7499.0,5.0
8,9,2026-05-05,C009,Odesa,Groceries,Card,NaN,0.0
9,10,2026-05-05,C010,Kharkiv,Books,Cash,340.0,0.0


In [36]:
transaction_df.head(5)

,transaction_id,transaction_date,customer_id,city,category,payment_method,amount,discount_percent
0,1,2026-05-01,C001,Kyiv,Books,Card,420.5,5.0
1,2,2026-05-01,C002,Lviv,Electronics,Card,3299.0,10.0
2,3,2026-05-02,C003,Odesa,Groceries,Cash,256.7,0.0
3,4,2026-05-02,C004,Kharkiv,Clothing,Card,1180.0,15.0
4,5,2026-05-03,C005,Kyiv,Groceries,Card,315.2,0.0


In [37]:
transaction_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 8 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   transaction_id    30 non-null     int64  
 1   transaction_date  30 non-null     str    
 2   customer_id       30 non-null     str    
 3   city              30 non-null     str    
 4   category          30 non-null     str    
 5   payment_method    30 non-null     str    
 6   amount            27 non-null     float64
 7   discount_percent  28 non-null     float64
dtypes: float64(2), int64(1), str(5)
memory usage: 2.0 KB


In [38]:
transaction_df.isna().sum()

transaction_id      0
transaction_date    0
customer_id         0
city                0
category            0
payment_method      0
amount              3
discount_percent    2
dtype: int64

In [40]:
transaction_df.describe()

,transaction_id,amount,discount_percent
count,30.000000,27.000000,28.000000
mean,15.500000,6475.281481,5.714286
std,8.803408,19445.919441,6.211442
min,1.000000,150.000000,0.000000
25%,8.250000,310.400000,0.000000
50%,15.500000,625.800000,5.000000
75%,22.750000,2489.500000,10.000000
max,30.000000,99999.000000,20.000000


7. Знайдіть найбільші значення `amount` і зробіть висновок, чи є серед них потенційні викиди. - є викид у вигляді максимального значення  99999.000000,  у вигляяді від'ємних значень викиди відсутні 


Медіана 625.800000  має таке знчення в зв'язку з однією дороговартісною покупкою, а mean	6475.281481	формується за рахунок кількості покупок


In [39]:
transaction_df['amount'].mode()

0       150.0
1       189.9
2       198.4
3       230.0
4       256.7
5       275.0
6       305.6
7       315.2
8       340.0
9       410.2
10      420.5
11      481.3
12      510.0
13      625.8
14      720.0
15      870.0
16      950.0
17     1180.0
18     1340.0
19     1680.0
20     3299.0
21     4599.0
22     7499.0
23     8990.0
24    15999.0
25    22999.0
26    99999.0
Name: amount, dtype: float64

In [41]:
transaction_df['amount'].quantile(0.1)   

np.float64(217.36)

In [42]:
transaction_df['amount'].quantile(0.9) 

np.float64(11793.600000000015)

In [43]:
transaction_df['amount'].quantile(0.95) 

np.float64(20898.999999999996)